tools/walletextension/subscriptions/subscriptions.go

```diff
@@ -106,39 +100,69 @@ func readFromChannelAndWriteToUserConn(channel chan common.IDAndLog, userConn us
 	}
 }
 
-func checkIfUserConnIsClosedAndUnsubscribe(userConn userconn.UserConn, subscription *gethrpc.ClientSubscription) {
-	for {
-		if userConn.IsClosed() {
-			subscription.Unsubscribe()
-			return
+func (sm *SubscriptionManager) unsubscribeAndRemove(userSubscriptionID string, subscription *gethrpc.ClientSubscription) {
+	sm.mu.Lock()
+	defer sm.mu.Unlock()
+
+	subscription.Unsubscribe()
+
+	subscriptions, exists := sm.subscriptionMappings[userSubscriptionID]
+	if !exists {
+		return
+	}
```

Comment:
Is this possible to happen ? that a subscription is cancelled but there isn't one in our mapping ?

Reply:
Should not happen, added it here to be 100% sure not to crash. I added a log that will clearly tell us if & what happened.